### PR TITLE
Add --gpus support

### LINKED ISSUE #5: Use hardware acceleration on the container
Allow the usage of hardware acceleration for complex tests and simulation

### PR DESCRIPTION
**Pull request by mateus-amarante**
## Description

Add --gpus support to test and build images

## Related Issues

Close #5

### CODE REVIEW

**Comment by mateus-amarante** [comment]
@ivan1993br , can we test the workflow_dispatch trigger if we merge https://github.com/open-br/ros_ws/pull/13 first?

File: run-devel.sh [hunk 1]
```diff
@@ -58,6 +58,8 @@ if [ ! "$(docker ps -q -f name=$CONTAINER)" ]; then
             --volume="/etc/sudoers.d:/etc/sudoers.d:ro" \
             --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" \
             --privileged \
+            --gpus all \
```
Comment (caioaamaral) [inline run-devel.sh]: Is a privileged container necessary for using gpu?

**Review by caioaamaral — APPROVED** [review]

File: run-devel.sh (hunk 1 above)
Comment (mateus-amarante) [reply run-devel.sh]: I don't know. However, the privileged is useful for other things, like accessing USB devices.

File: run-devel.sh [hunk 2]
```diff
@@ -58,6 +58,8 @@ if [ ! "$(docker ps -q -f name=$CONTAINER)" ]; then
...
+            --env="NVIDIA_DRIVER_CAPABILITIES=utility,graphics,compute" \
```
Comment (ivan1993br) [inline run-devel.sh]: How about:
'all,"capabilities=utility,graphics,compute"'?

File: run-test.sh [hunk 3]
```diff
@@ -51,6 +51,8 @@ docker run -it \
     --env ROS_DISTRO=$ROS_DISTRO \
     --volume $(realpath $SRC_REPOS):/root/src.repos \
     --volume="${UBUNTU_DISTRO}_apt_cache:/var/cache/apt/archives" \
+    --gpus all \
```
Comment (ivan1993br) [inline run-test.sh]: See previous comment

**Review by ivan1993br — APPROVED** [review]
Nice!

**Comment by ivan1993br** [comment]
> @ivan1993br , can we test the workflow_dispatch trigger if we merge https://github.com/open-br/ros_ws/pull/13 first?

Yes

File: run-devel.sh (hunk 2 above)
Comment (mateus-amarante) [reply run-devel.sh]: I tried something like this before but I had some problems. I can try again with your suggestions.

File: run-devel.sh (hunk 1 above)
Comment (caioaamaral) [reply run-devel.sh]: > I don't know. However, the privileged is useful for other things, like accessing USB devices.

Got it, but maybe we should keep the minimum requirement. I didn't look deeply, but apparently running privileged container is a [bad idea](https://www.trendmicro.com/en_us/research/19/l/why-running-a-privileged-container-in-docker-is-a-bad-idea.html). We can always set the specifics capabilities that we are in need, like [here](https://stackoverflow.com/a/36441605)

Comment (mateus-amarante) [reply run-devel.sh]: We can remove it but I would consider adding a parameter to the script to pass extra arguments. I also think it should come in another PR (create an issue first)

Comment (ivan1993br) [reply run-devel.sh]: I agree, for development, privileged will make our life easier during development, however, it is not safe

File: run-test.sh (hunk 3 above)
Comment (mateus-amarante) [reply run-test.sh]: Implemented in 820aa2c

File: run-devel.sh (hunk 2 above)
Comment (mateus-amarante) [reply run-devel.sh]: Implemented in 820aa2c

**Pull request merged**